modules/apps/transfer/keeper/relay_test.go

```diff
@@ -167,42 +254,83 @@ func (suite *KeeperTestSuite) TestOnRecvPacket() {
 		recvIsSource bool // the receiving chain is the source of the coin originally
 		expPass      bool
 	}{
-		{"success receive on source chain", func() {}, true, true},
-		{"success receive on source chain with memo", func() {
-			memo = "memo"
-		}, true, true},
-		{"success receive with coin from another chain as source", func() {}, false, true},
-		{"success receive with coin from another chain as source with memo", func() {
-			memo = "memo"
-		}, false, true},
-		{"empty coin", func() {
-			trace = types.DenomTrace{}
-			amount = sdk.ZeroInt()
-		}, true, false},
-		{"invalid receiver address", func() {
-			receiver = "gaia1scqhwpgsmr6vmztaa7suurfl52my6nd2kmrudl"
-		}, true, false},
+		{
+			"success receive on source chain",
+			func() {}, true, true,
+		},
+		{
+			"success receive on source chain of half the amount",
+			func() {
+				amount = math.NewInt(50)
+				expEscrowAmount = math.NewInt(50)
```

Comment:
If I send back the full amount I wouldn't be sure of whether the subtraction happened correctly when the tokens are sent back or something else is wrong in the code, since there could be a situation where we have a bug in setting the escrow amount both on send and recv and the result would be the same as if it worked (the total amount in escrow at the end if 0).